### PR TITLE
refactor `chat_id` in `MessegerService`

### LINKED ISSUE #57: Modify MessegerService to receive Chat ID on creation
Possibly also context.
That way, can use messenger.send(message) and avoid passing chat_id to the messenger, and make get_messenger_and_chat_id simpler -> just make_messenger().

### PR DESCRIPTION
**Pull request by nnrepos**
resolved #57.

## changes
* added member `chat_id` in the `MessegerService` class.
* refactored usage of `MessegerService` class.

## personal user experience
* the `INSTALL` file didn't work for me (some zlib and pip issues), i ended up asking chatgpt and it was fine :)
* it's a good practice to add a `CONTRIBUTING.md` file to open source projects, that's usually the first thing newcomers look for.
* as described in the issue, it's difficult running the tests - please consider adding git actions on PR creation - i doubt the API usage will be abused.
* i would also add an auto pep8 check. i do not recommend setting up mypy for early projects - it usually requires too much work. 
* please consider adding a template for pull requests.
* other than that, the project looks nice - i look forward to creating more PRs.

### CODE REVIEW

**Comment by yairl** [comment]
Thanks!

Really excited to see your PR.
Out on a family vacation, expect PR to be handled tomorrow.

File: src/message_handler.py [hunk 1]
```diff
@@ -62,8 +62,9 @@ def handle_incoming_message_core(ctx:Context, event, in_flight):
     start = time.time()
     parsed_event = json.loads(event)
     ctx.log(parsed_event)
-    messenger = messenger_factory.messenger_by_type[parsed_event["source"]]
-    
+    chat_id = str(parsed_event["event"]['message']['chat']['id'])
```
Comment (yairl) [inline src/message_handler.py:65]: This code would work for Telegram, but not WhatsApp.

Note how chat_id is set in wa.py; ideally, add another variant of messenger_factory.make_messenger(raw_message) or perhaps messenger_factory.make_messenger(source, message).

There you can handle this correctly per source type (tg/wa).
As a side effect will also make this nicer, one-liner code.

**Review by yairl — COMMENTED** [review]
Code looks much nicer after your changes.
Thanks!

There's one apparent bug; please check the relevant comment.

I will start integrating most of your "personal experience" comments separately.

Thanks!

File: src/message_handler.py (hunk 1 above)
Comment (nnrepos) [reply src/message_handler.py]: yep, i missed that.
i've committed a fix, even though it's not as concise as i'd want it to be... whatsapp's event kind complicates things a bit.

File: src/services/messengers/messenger_factory.py [hunk 2]
```diff
@@ -1,12 +1,37 @@
...
+
+def make_messenger(messenger_chat_id: str) -> MessagingService:
     messenger_str, chat_id = messenger_chat_id.split(":")
-    messenger = messenger_by_type[messenger_str]
-    return messenger, chat_id
+    messenger = messenger_by_type[messenger_str](chat_id)
+    return messenger
+
+
+def _make_wa_messenger_from_event(event: Dict) -> Optional[MessagingService]:
```
Comment (yairl) [inline src/services/messengers/messenger_factory.py:15]: A bit ugly, probably would've been better to have:

chat_id = <>
make_messenger(type, chat_id)

For all functions, and then:

make_messenger_from_event
make_messenger_from_<> (for the other one)

but good enough for now :)

**Review by yairl — APPROVED** [review]

**Pull request merged**